framework/src/main/java/org/checkerframework/framework/util/JavaExpressionParseUtil.java

```diff
@@ -522,51 +522,84 @@ public JavaExpression visit(NameExpr expr, JavaExpressionContext context) {
         }
 
         /**
-         * If {@code identifier} is a field name, then return the {@link FieldAccess} corresponding
-         * to using that field at the given {@code context}. If {@code identifier} is not a field
-         * name, this method returns null.
+         * Return the {@link FieldAccess} expression for the field with name {@code identifier}
+         * accessed via {@code receiverExpr}. If no such field exists, then {@code null} is
+         * returned.
          *
-         * @param receiverExpr the receiver of the field; the expression used to access the field
+         * @param receiverExpr the receiver of the field access; the expression used to access the
+         *     field
          * @param identifier possibly a field name
-         * @return a field access, or null if {@code identifier} is not a field
+         * @return a field access, or null if {@code identifier} is not a field that can be accessed
+         *     via {@code receiverExpr}
          */
         protected @Nullable FieldAccess getIdentifierAsField(
                 JavaExpression receiverExpr, String identifier) {
-            TypeMirror receiverType = receiverExpr.getType();
-            // isOriginalReceiver is true if receiverType has not been reassigned.
-            boolean isOriginalReceiver = true;
-            VariableElement fieldElem = null;
-            if (identifier.equals("length") && receiverType.getKind() == TypeKind.ARRAY) {
-                fieldElem = resolver.findField(identifier, receiverType, pathToCompilationUnit);
-            }
-            if (fieldElem == null) {
+            // Find the field element.
+            TypeMirror enclosingTypeOfField = receiverExpr.getType();
+            VariableElement fieldElem;
+            if (identifier.equals("length") && enclosingTypeOfField.getKind() == TypeKind.ARRAY) {
+                fieldElem =
+                        resolver.findField(identifier, enclosingTypeOfField, pathToCompilationUnit);
+                if (fieldElem == null) {
```

Comment:
How can this happen?  Every array type should have a length field.  If this test is retained, I think it should throw BugInCF rather than returning null.